beluga/include/beluga/algorithm/estimation.hpp

```diff
@@ -102,6 +161,79 @@ Sophus::Matrix2<Scalar> calculate_covariance(Range&& range, const Sophus::Vector
       mean);
 }
 
+/// Returns a pair consisting of the estimated mean pose and its covariance.
+/**
+ * Given a range of poses, computes the estimated pose by averaging the translation
+ * and rotation parts.
+ * Computes the covariance matrix of the translation and rotation parts to create a 6x6 covariance matrix.
+ * NOTE: The covariance matrix is in se3, the tangent space of the SE3 manifold. Users may perform
+ * the appropriate conversions to get the covariance matrix into their parametrization of interest.
```

Comment:
@serraramiro1 consider expanding a bit, explaining the rationale for the choice, plus a reference to relevant literature.